### PR TITLE
Samurai ci

### PR DESCRIPTION
**Pull request by supreethms1809**
Samurai CPU CI using Docker. This github action runs on github virtual machine.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    cmake .. && \
+    make -j 2
+
+ENV PATH="/app/samurai/build:${PATH}"
+ENV LD_LIBRARY_PATH="/usr/local/lib:${LD_LIBRARY_PATH}"
+ENV PATH="/usr/local/bin:${PATH}"
+
+RUN echo "#!/bin/bash \n\
+cd /app/samurai/build/release/bin/ \n\
+./samurai /app/data/beltrami_preprocessed.xml\
```
Comment (johnmauff) [inline Dockerfile:25]: Samurai is now using the TDRP configuration files.  I have placed the TDRP files in the ncar_scripts/TDRP directory.  The equivalent command to run the beltrami testcase is now the following 

./samurai -params beltrami.tdrp

**Review by johnmauff — COMMENTED** [review]
@supreethms1809 I have included some comments about the changes on how to run the code.  I still don't know how to explicitly run these tests myself.  It would also be nice to be able to run the other test cases at less frequent intervals.  I assume that this would require the creation of a new container that has additional datasets.  The other test cases are:  

supercell hurricane typhoonChanthu2020

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+FROM ssuresh1809/samurai:latest
+
+LABEL maintainer="Supreeth Suresh <ssuresh@ucar.edu>"
+
+RUN mkdir -p /app/
+WORKDIR /app/
+RUN git clone https://github.com/mmbell/samurai.git && \
+    cd samurai && \
+    sed -i 's/MODE GPU/MODE CPU/g' CMakeLists.txt && \
```
[the remark this answers is not in the document]
Comment (sjsprecious) [reply Dockerfile]: Done.

File: Dockerfile (hunk 1 above)
Comment (sjsprecious) [reply Dockerfile:26]: This will require Supreeth to upload the TDRP file to his image so that I could update the run command as suggested here.

**Comment by sjsprecious** [comment]
> @supreethms1809 I have included some comments about the changes on how to run the code. I still don't know how to explicitly run these tests myself. It would also be nice to be able to run the other test cases at less frequent intervals. I assume that this would require the creation of a new container that has additional datasets. The other test cases are:
> 
> supercell hurricane typhoonChanthu2020

I agree with John that we will need separate container images with different datasets and iteration counts. We could rename the image tags to `samurai_beltrami:cpu`, `samurai_supercell:cpu`, etc for clarification.

File: Dockerfile (hunk 1 above)
Comment (sjsprecious) [reply Dockerfile:26]: My bad. The data comes with SAMURAI source code and is not need to be copied manually.

**Comment by supreethms1809** [comment]
I am closing this PR. I have submitted another PR for this change

**Pull request closed without merging**